### PR TITLE
More doco

### PR DESCRIPTION
**Pull request by danni**

### CODE REVIEW

File: docs/aloe.rst [hunk 1]
```diff
@@ -30,4 +30,12 @@ extensively documented here.
 
     Run Python tests as well as Gherkin.
 
+.. option:: -a attr
+
+    Run features and scenarios with the given tag.
```
Comment (koterpillar) [inline docs/aloe.rst:35]: "Note: this is a Nose flag"?

File: docs/features.rst [hunk 2]
```diff
@@ -7,8 +7,100 @@ Writing Features
...
+----------
+
+The background_ is an optional section that is run before every scenario and
+contains steps. It is used to set up fixtures common to each scenario_ of
+the feature_.
+
+A background does not have a name or tags.
+
+If a step fails during the background all the scenarios of this feature_ will
+fail.
```
Comment (koterpillar) [inline docs/features.rst:37]: This is wrong. A (non-deterministic, which is bad) background can fail for one scenario and succeed for another.

File: docs/features.rst [hunk 3]
```diff
@@ -7,8 +7,100 @@ Writing Features
...
+                | Pets Inc           | Your one stop pet shop      |
+
+Scenario Outline
+----------------
+
+A scenario outline is a template for building scenarios from the rows of a
+table named ``Examples``. Parameters are written in the form ``<Parameter>``,
+where each named parameter must be present in the table.
+
+Scenario outlines have a name and my optionally have tags_.
```
Comment (koterpillar) [inline docs/features.rst:75]: may

File: docs/steps.rst [hunk 4]
```diff
@@ -6,6 +6,39 @@ Defining Steps
...
+
+**Number**
+
+    .. code-block:: gherkin
+
+        Then the price should be $12.99
+
+    .. code-block:: python
+
+        @step(r'The price should be \$(\d+\.?\d*)')
```
Comment (koterpillar) [inline docs/steps.rst:30]: That's not a good regular expression, it will uselessly backtrack. Consider `\d+(?:\.\d+)?`.

File: docs/steps.rst [hunk 5]
```diff
@@ -28,3 +61,107 @@ Tools for step writing
...
 
+Writing good BDD steps
+----------------------
+
+It's very easy with BDD testing to accidentally reinvent Python testing
+using a pseudo-language. Doing so removes much of the point of using BDD
+testing in the first place, so here is some advice to help write better BDD
+steps.
+
+1.  **Be declarative not imperative**
```
Comment (koterpillar) [inline docs/steps.rst:72]: Worthy goal, but this is not what's described here. The BDD steps are imperative by nature, compare: "I log on as danni" and "When I am logged in as danni".

What I think you mean is just not going too deep into the implementation details, perhaps the title should be "Avoid implementation details".

File: docs/steps.rst [hunk 6]
```diff
@@ -28,3 +61,107 @@ Tools for step writing
...
+
+    You can use :meth:`Step.behave_as` to write a step that chains up several
+    smaller steps.
+
+2.  **Avoid conjunctions in steps**
+
+    If you're writing a step that contains an `and` or other conjunction
+    consider breaking your step into two.
+
+    You can pass state between steps using :attr:`world`.
```
Comment (koterpillar) [inline docs/steps.rst:101]: Possible example: "Given I log out and log back in as danni"

File: docs/steps.rst [hunk 7]
```diff
@@ -28,3 +61,107 @@ Tools for step writing
...
+    This is especially true when setting up the initial state for a test.
+
+    Implementation:
+
+    .. code-block:: gherkin
+
+        Given the following flags are set:
+            | flags                      |
+            | user_registration_disabled |
+            | user_export_disabled       |
```
Comment (koterpillar) [inline docs/steps.rst:114]: Also add a suggestion of moving steps that are needed by _all_ tests into a `@before.each_example`.

File: docs/aloe.rst (hunk 1 above)
Comment (michelts) [inline docs/aloe.rst:35]: I agree with @danni. You can't assume lettuce users migrating to aloe are familiar with nose. If you don't want to describe the option here, you can mention it in the "porting from lettuce" section.

Comment (koterpillar) [inline docs/aloe.rst:35]: I agree too, I just want that text to be added in the documentation :)

Comment (michelts) [inline docs/aloe.rst:35]: :)

**Pull request merged**